cache/http.go

```diff
@@ -32,22 +32,50 @@ type httpCache struct {
 
 // NewHTTPCache ...
 func NewHTTPCache(cacheDir string, maxBytes int64, ensureSpacer EnsureSpacer) HTTPCache {
-	ensureCacheDir(cacheDir)
+	for _, subdir := range []string{"cas", "ac"} {
+		ensureDirExists(filepath.Join(cacheDir, subdir))
+	}
 	cache := NewCache(cacheDir, maxBytes)
 	loadFilesIntoCache(cache)
 	return &httpCache{cache, ensureSpacer, make(map[string]*sync.Mutex), &sync.Mutex{}}
 }
 
-func ensureCacheDir(path string) {
-	d, err := os.Open(path)
-	if err != nil {
-		err := os.MkdirAll(path, os.FileMode(0744))
+type artifactInfo struct {
+	hash       string
+	filePath   string // Absolute filesystem path
+	verifyHash bool // true for CAS items, false for AC items
+}
+
+// Parse cache artifact information from the request URL
+func artifactInfoFromUrl(url string, baseDir string) (*artifactInfo, error) {
```

Comment:
`url` is not really a URL but just the `path` component or did I miss something (reviewing on my iPad 😄).